himportscan/src/HImportScan/ImportScanner.hs

```diff
@@ -69,26 +65,55 @@ instance Aeson.ToJSON ModuleImport where
 scanImportsFromFile :: FilePath -> IO (Maybe ScannedImports)
 scanImportsFromFile filePath = do
   fileExists <- doesFileExist filePath
-  if fileExists
-  then Just . scanImports filePath <$> Text.readFile filePath
-  else pure Nothing
-
-scanImports :: FilePath -> Text -> ScannedImports
-scanImports filePath contents =
-  let preprocessedContents = Text.encodeUtf8 $ preprocessContents contents
-      sbuffer = case preprocessedContents of
-        PS ptr offset len -> StringBuffer ptr len offset
-      loc = mkRealSrcLoc (mkFastString filePath) 1 1
-   in case scanTokenStream filePath $ lexTokenStream sbuffer loc of
-    Left err -> error err
-    Right ScannedData{moduleName, importedModules, usesTH} ->
-      ScannedImports
-        { filePath = Text.pack filePath
-        , moduleName
-        , importedModules
-        , usesTH
-        }
-
+  if not fileExists
```

Comment:
Is the reversal of the if condition accidental?

Reply:
Kind of. I had swapped them over because the `then` was longer, but then I factored the `then` out. I'll switch it back.